JoinColony/colonyCDapp
# [Multi-sig] Update manage permissions saga to handle MultiSigRoleSet

## Description

[Block-ingestor counterpart](https://github.com/JoinColony/block-ingestor/pull/212)

This PR adds support to the 'Manage permissions' saga and block-ingestor handler to assign multi sig roles to users.

## Testing

1. Install the 'Multi-sig permissions' extension.
2. Run this query, any user who already has root permissions should be assigned root permissions with multi-sig. (Thanks to Bassgeta for implementing this on CDapp side in the previous PR, this PR adds the block-ingestor side to handle this and store it in the database).

```
query MyQuery {
  listColonyRoles(filter: {isMultiSig: {eq: true}}) {
    items {
      id
      isMultiSig
      role_0
      role_1
      role_2
      role_3
      role_5
      role_6
    }
  }
}
```

3. Create a new 'Manage permissions' action. The 'Take actions via Multi-sig' authority option will only show when the extension is installed.
4. Select a permissions type or select custom permissions.
5. Submit the action and run the query again to check the database for updates. Try adding / removing permissions from the same user to confirm the changes in the database are correct. ~~Please note the UI will need updating before it accurately reflects the selected permissions.~~ now implemented ~~Also note, role_6 (administration) is not used for multi-sig and will always be null.~~ not sure if this is actually the case, have put role_6 back in for now and will confirm.

## Diffs

**New stuff** ✨

* ColonyRole now has `isMultiSig` field and new id structure for multi-sig roles

**Changes** 🏗

* Manage permissions saga now handles multi sig roles
* Manage permissions block-ingestor handler now handles multi sig roles
* Manage permissions action sidebar UI changes and completed action UI changes

Resolves #2146 

<img width="466" alt="Screenshot 2024-04-26 at 10 05 17" src="https://github.com/JoinColony/colonyCDapp/assets/34915414/87754362-0f73-4c83-8116-29287f0b0388">

<img width="710" alt="Screenshot 2024-04-26 at 10 52 17" src="https://github.com/JoinColony/colonyCDapp/assets/34915414/9cf85e46-990b-4d43-ba38-d8501446bd50">
